pkg/datarepair/queue/queue_test.go

```diff
@@ -48,9 +45,9 @@ func TestDequeueEmptyQueue(t *testing.T) {
 
 		q := db.RepairQueue()
 
-		s, err := q.Dequeue(ctx)
-		assert.Error(t, err)
-		assert.Equal(t, pb.InjuredSegment{}, s)
+		s, err := q.Select(ctx)
+		require.Error(t, err)
+		require.Nil(t, s)
```

Comment:
I agree that the require.Nil check isn't necessary

Reply:
or actually, I think it's fine in this test case where otherwise we would use `_, err := q.Select(ctx)`. additionally I think we want to confirm that `Select` won't error but will return nil if there are no segments available.

Reply:
yes, it should be `_, err :=` if we remove the `require.Nil`.

>  additionally I think we want to confirm that Select won't error but will return nil if there are no segments available.

In case that there isn't such test case, it would be worth to add it.

Reply:
> additionally I think we want to confirm that Select won't error but will return nil if there are no segments available.

Checking the whole test cases, it looks if we follow the Go common conventions, the result can be ignore in case of an error as mentioned before, but we should check that the error is of the class `ErrEmptyQueue`

I did such change, see my commit https://github.com/storj/storj/pull/1717/commits/063dec403d0a74879d7d745e941b68e5f8a79b38 if it isn't OK, let me know and I will drop it.

Reply:
its a test.  its not important, but its okay to test that we're complying with conventions.  we probably should test that we're complying with conventions. :D

Reply:
Not very important, mostly I wanted to understand if we apply those conventions because if we always have to return `nil` when an error is returned, then we have to add those asserts on any test which check that.